pdpyras.py

```diff
@@ -30,6 +30,8 @@
 
 ITERATION_LIMIT = 1e4
 
+TIMEOUT = 5
```

Comment:
I think if you're creating a way to set a request timeout, it would also be good if this was exposed to be defined by the module user, as a variable.

Reply:
Hi @badnetmask,

I'm really sorry for the increased latency in updates on this project and I plan on devoting a few hours this week and weekend to a bit of upkeep.

I agree with your idea, but what you have so far is perfectly fine for now; thank you for contributing!

My only criticism is that the improvement could be made much fewer lines by modifying the `request` method, i.e. changing the keyword args sent to `requests.Session.request` (the `req_kw` dictionary):

https://github.com/PagerDuty/pdpyras/blob/b359a886961d956e1e578b60b993a2481c0ed2b9/pdpyras.py#L495-L500

Reply:
Thanks for the feedbacks, I'll update the code while taking into account both comments.

Reply:
> I think if you're creating a way to set a request timeout, it would also be good if this was exposed to be defined by the module user, as a variable.

What would the API for setting this timeout looks like in your opinion? And what kind of use-cases do you see?

I'm not sure it is worth adding a new variable to the constructor of both `APISession` and `EventsAPISession` classes to expose this timeout.

A user can still set `pdpyras.TIMEOUT` directly if really necessary.

The main use-case for this change is that we have a custom script using this library which got stuck forever when the network it was running on was attacked by a DDoS.

Reply:
A pattern I've loosely followed (and I won't say it's the best, but it works) is to let sessions be mutable objects and allow modifying their settings (i.e. retry / cooldown) by setting their attributes. That might help here.

While it's not very common for endpoints to take longer than 5 seconds to respond, there are a few (i.e. `/oncalls`) where this could potentially happen.

Reply:
Added this so that it'll get covered by retry logic:

https://github.com/PagerDuty/pdpyras/pull/49